src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,243 @@
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+
+      let users: User[] = [];
+
+      switch (args.options.role) {
+        case 'Owner':
+          users = await this.getOwners(args.options, groupId, logger);
+          break;
+        case 'Member':
+          users = await this.getMembers(args.options, groupId, logger);
+          break;
+        default:
+          const owners = await this.getOwners(args.options, groupId, logger);
+          const members = await this.getMembers(args.options, groupId, logger);
+
+          if (!args.options.properties) {
+            owners.forEach((owner: ExtendedUser) => {
+              for (let i = 0; i < members.length; i++) {
+                if (members[i].userPrincipalName === owner.userPrincipalName) {
+                  if (!owner.roles.includes('Member')) {
+                    owner.roles.push('Member');
+                  }
+                }
+              }
+            });
+          }
+
+          users = owners.concat(members);
+
+          if (!args.options.properties) {
+            users = users.filter((value, index, array) => index === array.findIndex(item => item.userPrincipalName === value.userPrincipalName));
+          }
+      }
+
+      await logger.log(users);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getGroupId(options: Options): Promise<string> {
+    const { groupId, groupDisplayName } = options;
+
+    if (groupId) {
+      return groupId;
+    }
+
+    return await aadGroup.getGroupIdByDisplayName(groupDisplayName!);
+  }
+
+  private async getOwners(options: Options, groupId: string, logger: Logger): Promise<ExtendedUser[]> {
+    const { properties, filter } = options;
+
+    if (this.verbose) {
+      await logger.logToStderr(`Retrieving owners of the group with id ${groupId}`);
+    }
+
+    const selectProperties: string = properties ?
+      `?$select=${properties.split(',').map(p => formatting.encodeQueryParameter(p.trim())).join(',')}` :
+      '?$select=id,displayName,userPrincipalName,givenName,surname';
+    const endpoint: string = `${this.resource}/v1.0/groups/${groupId}/owners${selectProperties}`;
+
+    let owners: ExtendedUser[] = [];
+
+    if (filter) {
+      // While using the filter, we need to specify the ConsistencyLevel header.
+      const requestOptions: CliRequestOptions = {
+        url: `${endpoint}&$filter=${encodeURIComponent(filter)}&$count=true`,
+        headers: {
+          accept: 'application/json;odata.metadata=none',
+          ConsistencyLevel: 'eventual'
+        },
+        responseType: 'json'
+      };
+
+      owners = await odata.getAllItems<ExtendedUser>(requestOptions);
+    }
+    else {
+      owners = await odata.getAllItems<ExtendedUser>(endpoint);
+    }
+
+    if (!properties) {
```

Comment:
This if-statement should be removed as well. We know it has the owner or member role, based on the endpoint we are querying.